src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -603,26 +603,29 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
             }
         }
 
-        internal void EnqueueItem(IDocument doc, AnalysisPriority priority = AnalysisPriority.Normal, bool enqueueForAnalysis = true) {
+        internal void EnqueueItem(IDocument doc, AnalysisPriority priority = AnalysisPriority.Normal, bool parse = true, bool enqueueForAnalysis = true) {
             _disposableBag.ThrowIfDisposed();
             var pending = _pendingAnalysisEnqueue.Incremented();
             try {
-                Task<IAnalysisCookie> cookieTask;
-                using (GetDocumentParseCounter(doc, out var count)) {
-                    if (count > 3) {
-                        // Rough check to prevent unbounded queueing. If we have
-                        // multiple parses in queue, we will get the latest doc
-                        // version in one of the ones to come.
-                        return;
-                    }
-                    TraceMessage($"Parsing document {doc.DocumentUri}");
-                    cookieTask = ParseQueue.Enqueue(doc, Analyzer.LanguageVersion);
-                }
-
                 if (doc is ProjectEntry entry) {
                     entry.ResetCompleteAnalysis();
                 }
 
+                // If we don't need to parse, use null cooking
+                var cookieTask = Task.FromResult<IAnalysisCookie>(null);
```

Comment:
If cookie is null, `EditorFile.UpdateParseDiagnostics` won't be called. Is that expected?

Reply:
Yes, since we didn't reparse